DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -151,13 +94,10 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
                 throw new DeviceRequiredPropertyNotFoundException("'DeviceID' property is missing");
             }
 
-            string deviceId = device.DeviceProperties.DeviceID;
-
-            DeviceModel existingDevice = await this.GetDeviceAsync(deviceId);
-
+            DeviceModel existingDevice = await GetDeviceAsync(device.id);
             if (existingDevice == null)
             {
-                throw new DeviceNotRegisteredException(deviceId);
+                throw new DeviceNotRegisteredException(device.id);
```

Comment:
No this is a bug. That should DeviceNotRegisteredException(device.DeviceProperties.ID)